ForMaL/Workshops/workshop1.lean

```diff
@@ -0,0 +1,111 @@
+/- # Basics: rewriting, calculating, reasoning forwards and backwards
+References: [MIL] Mathematics in Lean, [Tut] Tutorials project.
+Most of the demonstration section comes [MIL].
+-/
+
+import Mathlib.Tactic
+
+/-
+# Rewriting and calculating
+-/
+
+-- Lean knows some lemmas about real numbers.
+#check mul_comm
+#check mul_assoc
+
+-- The rewrite tactic can use these lemmas.
+example (a b c : ℝ) : a * b * c = b * (a * c) := by
+  rw [mul_comm a b]
+  rw [mul_assoc] -- no need to supply variables here
+
+-- Can also rewrite backwards, and use hypotheses.
+example (a b c d e f : ℝ) (h : a * b = c * d) (h' : e = f) : a * (b * e) = c * (d * f) := by
+  rw [h']
+  rw [← mul_assoc]
+  rw [h]
+  rw [mul_assoc]
+
+-- A calc block can help to structure a sequence of rewrites.
+example : (a + b) * (a + b) = a * a + 2 * (a * b) + b * b :=
+  calc
+    (a + b) * (a + b) = a * a + b * a + (a * b + b * b) := by
+      rw [mul_add, add_mul, add_mul]
```

Comment:
add another example before this one, showing that you can combine the rewrites together